lib/improver/metadata/set_attributes.py

```diff
@@ -0,0 +1,144 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""
+Module setting attributes for IMPROVER blended data, nowcasts and spot
+forecasts
+"""
+
+import re
+import warnings
+
+from improver.metadata.constants.attributes import (
+    STANDARD_GRID_TITLE_STRING,
+    UK_SPOT_TITLE_STRING,
+    GLOBAL_SPOT_TITLE_STRING,
+    DATASET_ATTRIBUTES)
+from improver.utilities.cube_metadata import amend_metadata
+
+
+def set_product_attributes(cube, product):
+    """
+    Set attributes on an output cube of type matching a key string in the
+    DATASET_ATTRIBUTES dictionary.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube containing product data
+        product (str):
+            String describing product type
+
+    Returns:
+        updated_cube (iris.cube.Cube):
+            Cube with updated attributes
+    """
+    try:
+        original_title = cube.attributes["title"]
+    except KeyError:
+        original_title = ""
+
+    try:
+        updated_cube = amend_metadata(
+            cube, attributes=DATASET_ATTRIBUTES[product])
+    except KeyError:
+        options = list(DATASET_ATTRIBUTES.keys())
+        raise ValueError(
+            "product '{}' not available (options: {})".format(
+                product, options))
+
+    if STANDARD_GRID_TITLE_STRING in original_title:
+        updated_cube.attributes["title"] += " on {}".format(
+            STANDARD_GRID_TITLE_STRING)
+
+    return updated_cube
+
+
+def _match_title(original_title):
+    """
+    Match a title string to a regular expression
+
+    Args:
+        original_title (str):
+            Value of "title" attribute on an input cube
+
+    Returns:
+        match (str or None):
```

Comment:
The output here is actually a [regular expression object](https://docs.python.org/3/library/re.html#regular-expression-objects), rather than a string. If I do type(regex.match(original_title)), I get `<class '_sre.SRE_Match'>`.

Reply:
That explains a few things!  I'll correct this docstring, thanks.